src/controllers/edgeconnect/version/edgeConnect_test.go

```diff
@@ -0,0 +1,137 @@
+package version
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	edgeconnectv1alpha1 "github.com/Dynatrace/dynatrace-operator/src/api/v1alpha1/edgeconnect"
+	"github.com/Dynatrace/dynatrace-operator/src/registry"
+	"github.com/Dynatrace/dynatrace-operator/src/scheme/fake"
+	"github.com/Dynatrace/dynatrace-operator/src/timeprovider"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/opencontainers/go-digest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const fakeDigest = "sha256:7173b809ca12ec5dee4506cd86be934c4596dd234ee82c0662eac04a8c2c71dc"
+
+type registryClientMock struct {
```

Comment:
a mock for this already exists: https://github.com/Dynatrace/dynatrace-operator/blob/c1b3de905ad44a3e82b4204ac6806ce862b51b5d/src/registry/mocks/client.go#L15

(you would still need to set it up)

Reply:
I did this as it is easier to read if just the interface is implemented and one can immediately see what the getimageversion is doing

Reply:
it would be the same level of readable (just create a "setupMockedGetter", where you setup the mock), but you would have more tools to check what is happening, and the actual mock would catch unexpected calls.

but, as its a simple interface I will consider this a nitpick :D